tests/Events.elm

```diff
@@ -20,62 +21,62 @@ all =
+                    |> Event.expect SampleMsg
         , test "returns msg for click on lazy html" <|
             \() ->
                 Query.fromHtml sampleLazyHtml
                     |> Query.findAll [ tag "button" ]
                     |> Query.first
-                    |> Events.simulate Click
-                    |> Events.expectEvent SampleMsg
+                    |> Event.simulate Event.click
+                    |> Event.expect SampleMsg
         , test "returns msg for click on mapped html" <|
             \() ->
                 Query.fromHtml sampleMappedHtml
                     |> Query.findAll [ tag "button" ]
                     |> Query.first
-                    |> Events.simulate Click
-                    |> Events.expectEvent MappedSampleMsg
+                    |> Event.simulate Event.click
+                    |> Event.expect MappedSampleMsg
         , test "returns msg for click on deep mapped html" <|
             \() ->
                 Query.fromHtml deepMappedHtml
                     |> Query.findAll [ tag "input" ]
                     |> Query.first
-                    |> Events.simulate (Input "foo")
-                    |> Events.expectEvent (SampleInputMsg "foobar")
+                    |> Event.simulate (Event.input "foo")
+                    |> Event.expect (SampleInputMsg "foobar")
         , test "returns msg for input with transformation" <|
             \() ->
                 input [ onInput (String.toUpper >> SampleInputMsg) ] []
                     |> Query.fromHtml
-                    |> Events.simulate (Input "cats")
-                    |> Events.expectEvent (SampleInputMsg "CATS")
+                    |> Event.simulate (Event.input "cats")
+                    |> Event.expect (SampleInputMsg "CATS")
         , test "returns msg for check event" <|
             \() ->
                 input [ onCheck SampleCheckedMsg ] []
                     |> Query.fromHtml
-                    |> Events.simulate (Check True)
-                    |> Events.expectEvent (SampleCheckedMsg True)
+                    |> Event.simulate (Event.check True)
+                    |> Event.expect (SampleCheckedMsg True)
         , test "returns msg for custom event" <|
             \() ->
                 input [ on "keyup" (Json.Decode.map SampleKeyUpMsg keyCode) ] []
                     |> Query.fromHtml
-                    |> Events.simulate (CustomEvent "keyup" "{\"keyCode\": 5}")
-                    |> Events.expectEvent (SampleKeyUpMsg 5)
-        , testEvent onDoubleClick DoubleClick
-        , testEvent onMouseDown MouseDown
-        , testEvent onMouseUp MouseUp
-        , testEvent onMouseLeave MouseLeave
-        , testEvent onMouseOver MouseOver
-        , testEvent onMouseOut MouseOut
-        , testEvent onSubmit Submit
-        , testEvent onBlur Blur
-        , testEvent onFocus Focus
+                    |> Event.simulate ( "keyup", Encode.object [ ( "keyCode", Encode.int 5 ) ] )
```

Comment:
should we add a custom event function?  Because then we can add documentation to it. Otherwise there is no documentation on how to create custom events.